packages/react-components/src/components/VideoGallery/LargeGalleryLayout.tsx

```diff
@@ -0,0 +1,185 @@
+   * re-render the initial tiles -> dispose them -> create new tiles, we need to take care of
+   * this case when those components are here
+   */
+  const [indexesToRender, setIndexesToRender] = useState<number[]>([]);
+
+  const overflowGalleryTiles = overflowGalleryParticipants.map((p, i) => {
+    return onRenderRemoteParticipant(
+      p,
+      maxRemoteVideoStreams && maxRemoteVideoStreams >= 0
+        ? p.videoStream?.isAvailable && indexesToRender.includes(i) && activeVideoStreams++ < maxRemoteVideoStreams
+        : p.videoStream?.isAvailable
+    );
+  });
+
+  if (localVideoComponent) {
+    gridTiles.push(localVideoComponent);
+  }
+
+  const overflowGallery = useMemo(() => {
+    if (overflowGalleryTiles.length === 0) {
+      return null;
+    }
+    return (
+      <OverflowGallery
+        isNarrow={isNarrow}
+        /* @conditional-compile-remove(gallery-layouts) */
+        isShort={isShort}
+        shouldFloatLocalVideo={false}
+        overflowGalleryElements={overflowGalleryTiles}
+        horizontalGalleryStyles={styles?.horizontalGallery}
+        /* @conditional-compile-remove(vertical-gallery) */
+        verticalGalleryStyles={styles?.verticalGallery}
+        /* @conditional-compile-remove(vertical-gallery) */
+        overflowGalleryPosition={overflowGalleryPosition}
+        onFetchTilesToRender={setIndexesToRender}
+        onChildrenPerPageChange={(n: number) => {
+          childrenPerPage.current = n;
+        }}
+      />
+    );
+  }, [
+    isNarrow,
+    /* @conditional-compile-remove(gallery-layouts) */ isShort,
+    overflowGalleryTiles,
+    styles?.horizontalGallery,
+    /* @conditional-compile-remove(vertical-gallery) */ overflowGalleryPosition,
+    setIndexesToRender,
+    /* @conditional-compile-remove(vertical-gallery) */ styles?.verticalGallery
+  ]);
+
+  return (
+    <Stack
+      /* @conditional-compile-remove(vertical-gallery) */
+      horizontal={overflowGalleryPosition === 'VerticalRight'}
+      styles={rootLayoutStyle}
+      tokens={videoGalleryLayoutGap}
+    >
+      {
+        /* @conditional-compile-remove(gallery-layouts) */ props.overflowGalleryPosition === 'HorizontalTop' ? (
+          overflowGallery
+        ) : (
+          <></>
+        )
+      }
+      {screenShareComponent ? (
+        screenShareComponent
+      ) : (
+        <GridLayout key="grid-layout" styles={styles?.gridLayout}>
+          {gridTiles}
+        </GridLayout>
+      )}
+      {overflowGalleryTrampoline(
+        overflowGallery,
+        /* @conditional-compile-remove(gallery-layouts) */ props.overflowGalleryPosition
+      )}
+    </Stack>
+  );
+};
+
+const overflowGalleryTrampoline = (
+  gallery: JSX.Element | null,
+  galleryPosition?: 'HorizontalBottom' | 'VerticalRight' | 'HorizontalTop'
+): JSX.Element | null => {
+  /* @conditional-compile-remove(gallery-layouts) */
+  return galleryPosition !== 'HorizontalTop' ? gallery : <></>;
```

Comment:
we don't show a overflow gallery if its horizontal top?

Reply:
Oh its to do with placement of the gallery, avoid putting one check for HorizontalTop in a callback function, but the other just inline:
![image](https://github.com/Azure/communication-ui-library/assets/2684369/9c5a409d-c3df-464b-b6ff-232fe715c5ff)

Reply:
Out of interest, instead of having to do this check and having the overflow sorta defined in two places, can you set `reversed` property on the stack so children are in the opposite order (so `reversed={galleryPosition === 'HorizontalTop'}`) -- not sure how it would impact the a11y keyboard ordering though

Reply:
there is a bug with the fluent stacks where that wont work sadly, as for the callback its just for CC reasons since vertical gallery is also not GA'd, that is part of this effort